src/servers/public-api/fines/fines.controller.ts

```diff
@@ -49,9 +49,16 @@ export class FinesController {
   ) {
     try {
       const { loanID } = request.params;
-      const fine = await services.fines.searchFine({
-        loanId: Number(loanID),
-      });
+      let page = Number(request.query.page);
+      let limit = Number(request.query.limit);
+      const startIndex = (page - 1) * limit;
```

Comment:
This shouldn't be done here. 

Remember, services are meant to be reusable throughout the codebase, doing this here means that you'll have to do it else where that's invoking the method. 

Please pass the pagination (page, limit) into the service and determine the offset and limit there instead